### PR TITLE
[GLUTEN-6681] [CH]fix array(decimal32) in CH columnar to row

### PR DESCRIPTION
**Pull request by loudongfeng**
## What changes were proposed in this pull request?

(Fixes: \#6681)

## How was this patch tested?

by UT

### CODE REVIEW

File: cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp [hunk 1]
```diff
@@ -702,6 +701,12 @@ int64_t VariableLengthDataWriter::writeArray(size_t row_idx, const DB::Array & a
                     auto v = elem.get<Float64>();
                     writer.unsafeWrite(reinterpret_cast<const char *>(&v), buffer_address + offset + start + 8 + len_null_bitmap + i * elem_size);
                 }
+                else if (writer.getWhichDataType().isDecimal32())
+                {
+                  // We can not use get<char>() directly here to process Decimal32 field,
+                  // because it will get 4 byte data, but Decimal32 is 8 byte in Spark, which will cause error conversion.
+                  writer.write(elem, buffer_address + offset + start + 8 + len_null_bitmap + i * elem_size);
```
[the remark this answers is not in the document]
Comment (loudongfeng) [reply cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp:708]: Tried "SELECT struct(1.0, 2.0)" ,did not find any issue.It already use 8 bytes to write Decimal32.

File: cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp [hunk 2]
```diff
@@ -702,6 +701,12 @@ int64_t VariableLengthDataWriter::writeArray(size_t row_idx, const DB::Array & a
                     auto v = elem.get<Float64>();
                     writer.unsafeWrite(reinterpret_cast<const char *>(&v), buffer_address + offset + start + 8 + len_null_bitmap + i * elem_size);
                 }
+                else if (writer.getWhichDataType().isDecimal32())
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp:704]: 这个改动也影响了 spark row to ch column, 最好也能加个ut.

**Review by taiyang-li — CHANGES_REQUESTED** [review]

File: cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp (hunk 2 above)
Comment (loudongfeng) [reply cpp-ch/local-engine/Parser/CHColumnToSparkRow.cpp:704]: 加了一个用例

**Review by taiyang-li — APPROVED** [review]
LGTM

**Pull request merged**